src/EventListener/ProductTranslationUpdateListener.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusRedirectPlugin\EventListener;
+
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ORM\EntityManagerInterface;
+use Setono\SyliusRedirectPlugin\Factory\RedirectFactoryInterface;
+use Setono\SyliusRedirectPlugin\Finder\RemovableRedirectFinderInterface;
+use Sylius\Bundle\ResourceBundle\Event\ResourceControllerEvent;
+use Sylius\Component\Core\Model\Product;
+use Sylius\Component\Core\Model\ProductTranslationInterface;
+use Sylius\Component\Product\Model\ProductInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Validator\ConstraintViolationInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+final class ProductTranslationUpdateListener
+{
+    /** @var \Symfony\Component\HttpFoundation\Request|null */
+    private $request;
+
+    /** @var RedirectFactoryInterface */
+    private $redirectFactory;
+
+    /** @var ValidatorInterface */
+    private $validator;
+
+    /** @var RemovableRedirectFinderInterface */
+    private $removableRedirectFinder;
+
+    /** @var array */
+    private $validationGroups;
+
+    /** @var \Doctrine\ORM\EntityManagerInterface */
+    private $entityManager;
+
+    public function __construct(RequestStack $requestStack,
+                                RedirectFactoryInterface $redirectFactory,
+                                ValidatorInterface $validator,
+                                ManagerRegistry $managerRegistry,
+                                RemovableRedirectFinderInterface $removableRedirectFinder,
+                                array $validationGroups
+    ) {
+        $this->request = $requestStack->getCurrentRequest();
```

Comment:
It's best practice to first call things like this when needed. You could have a protected method for that :)